src/solvers/smt/smt_conv.cpp

```diff
@@ -2367,6 +2424,16 @@ smt_astt smt_convt::array_create(const expr2tc &expr)
 {
   if(is_constant_array_of2t(expr))
     return convert_array_of_prep(expr);
+  // Check size
+  assert(is_constant_array2t(expr) || is_constant_vector2t(expr));
+  expr2tc size = is_vector_type(expr) ? to_vector_type(expr->type).array_size
+                                      : to_array_type(expr->type).array_size;
+  bool is_infinite = is_vector_type(expr)
+                       ? to_vector_type(expr->type).size_is_infinite
+                       : to_array_type(expr->type).size_is_infinite;
+  auto members = is_vector_type(expr)
+                   ? to_constant_vector2t(expr).datatype_members
+                   : to_constant_array2t(expr).datatype_members;
```

Comment:
You could use the common base class of both types to simplify these 3 assignments.

Reply:
`Constant_vector` and `constant_array` do not share a common class.

Reply:
The comment on both classes says
```c
@extends constant_datatype_data
```

Reply:
Well, and the types share `@extends array_data` :) I think we could get rid of all these `?:` here.

Reply:
Ow, you are correct. Replaced them with static_pointer_cast